extensions/powershell/cmdlet-class.ts

```diff
@@ -501,10 +501,14 @@ export class CmdletClass extends Class {
             const unexpected = function* () {
               yield `// Unrecognized Response. Create an error record based on what we have.`;
               if (each.details.csharp.responseType) {
-                yield `WriteError(new global::System.Management.Automation.ErrorRecord(new ${ClientRuntime.name}.RestException<${each.details.csharp.responseType}>(responseMessage, await response), responseMessage.StatusCode.ToString(), System.Management.Automation.ErrorCategory.InvalidOperation, new {  ${operationParameters.filter(e => valueOf(e.expression) !== 'null').map(each => `${each.name}=${each.expression}`).join(',')}}));`;
+                yield `var ex = new ${ClientRuntime.name}.RestException<${each.details.csharp.responseType}>(responseMessage, await response);`;
+
               } else {
-                yield `WriteError(new global::System.Management.Automation.ErrorRecord(new ${ClientRuntime.name}.RestException(responseMessage), responseMessage.StatusCode.ToString(), System.Management.Automation.ErrorCategory.InvalidOperation, new {  ${operationParameters.filter(e => valueOf(e.expression) !== 'null').map(each => `${each.name}=${each.expression}`).join(',')}}));`;
+                yield `var ex = new ${ClientRuntime.name}.RestException(responseMessage);`
```

Comment:
I'm not 100% clear on whether I should be using Local("ex", `blah`) for this or not. This works, but I can change it if it goes against some best practice.

Reply:
It's the better practice (I wrote the `Local(...)` stuff later, after I had a bunch of code written, and haven't gone back to clean it all up yet